### DISCUSSION TITLE
Loop Error

### DISCUSSION THREAD

**Discussion by CICABERR** [seleniumbase/SeleniumBase#2867] [Q&A] [answered]
Hello, I have 51 `<a>` elements on a website from which I want to extract the link addresses. I select these elements using the code line `title_element = sb.find_elements(By.CLASS_NAME, "classifiedTitle")`. However, when I perform operations within a `for` loop, I encounter an error on the second iteration specifically when selecting the `<a>` element in the loop. When I simply print the links within the loop without using `click()`, all 51 links are printed correctly and successfully. However, when I use `click()` and `go_back()` within the loop, I encounter an error on the second iteration. Where could I be making a mistake? Can you help me with this issue?

`title_element = sb.find_elements(By.CLASS_NAME, "classifiedTitle")
        print(len(title_element))
        for i in range(len(title_element)):
            print(i)
            print(title_element[i]) 
            href_value = title_element[i].get_attribute("href")
            print(href_value)
            sb.execute_script("arguments[0].click();", title_element[i])
            sb.wait_for_ready_state_complete()
            sb.go_back()
            sb.wait_for_ready_state_complete()`
![selenium error](https://github.com/seleniumbase/SeleniumBase/assets/154458060/51adbce6-60ef-400a-9f5e-3d50cd13ebf8)

**Comment by mdmintz** [comment, marked as answer]
If you find an element, leave the website, and then come back to the website, then that element is "Stale", and you'll get a "StaleElementReferenceException" if you try to use it. Elements that you find can only be used if you don't leave the website that you found them on. Build your scripts accordingly to avoid running into that scenario.
